lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -0,0 +1,50 @@
+'use strict'
+
+/**
+ * TODO: Support batch queries
+ */
+
+var semver = require('semver')
+var sqlSummary = require('sql-summary')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (cassandra, agent, version, enabled) {
+  if (!enabled) return cassandra
+  if (!semver.satisfies(version, '^3.0.0')) {
+    agent.logger.debug('cassandra version %s not supported - aborting...', version)
+    return cassandra
+  }
+
+  agent.logger.debug('shimming cassandra.Client.prototype._innerExecute')
+  shimmer.wrap(cassandra.Client.prototype, '_innerExecute', wrapInnerExecute)
+
+  return cassandra
+
+  function wrapInnerExecute (original) {
+    return function wrappedInnerExecute (...args) {
+      var span = agent.buildSpan()
+      var id = span && span.transaction.id
+
+      if (span) {
+        var query = args[0]
+        span.type = 'db.cassandra.query'
+        span.setDbContext({ statement: query, type: 'sql' })
```

Comment:
I've never used Cassandra, so excuse my ignorance here... The query language looks like SQL, but they seem to call it CQL. But is it just an alias?

Reply:
I just checked with the Python agent and they call it `sql` as well, but I think that might be wrong. I just checked what done in OpenTracing. I found one semi official Cassandra instrumentation and they use type `cassandra`:

https://github.com/opentracing-contrib/java-cassandra-driver/blob/a3d068c8601146dfdaf46557422a55379505d9fb/src/main/java/io/opentracing/contrib/cassandra/TracingSession.java#L319

Reply:
I changed it to `cassandra`.